code/modules/mob/living/carbon/human/life.dm

```diff
@@ -23,33 +23,31 @@
 		return
 
 	. = ..()
+
 	if(QDELETED(src))
 		return FALSE
 
-	//Body temperature stability and damage
+	// Body temperature stability and damage
 	dna.species.handle_body_temperature(src, seconds_per_tick, times_fired)
-	if(!HAS_TRAIT(src, TRAIT_STASIS))
-		if(stat != DEAD)
-			//handle active mutations
-			for(var/datum/mutation/mutation as anything in dna.mutations)
-				mutation.on_life(seconds_per_tick, times_fired)
-			//heart attack stuff
-			handle_heart(seconds_per_tick, times_fired)
-			//handles liver failure effects, if we lack a liver
-			handle_liver(seconds_per_tick, times_fired)
-
-		// for special species interactions
-		dna.species.spec_life(src, seconds_per_tick, times_fired)
-	else
+	if(HAS_TRAIT(src, TRAIT_STASIS))
 		for(var/datum/wound/iter_wound as anything in all_wounds)
 			iter_wound.on_stasis(seconds_per_tick, times_fired)
+		return stat != DEAD
 
-	//Update our name based on whether our face is obscured/disfigured
-	name = get_visible_name()
-
-	if(stat != DEAD)
-		return TRUE
+	if(stat == DEAD)
+		return FALSE
 
+	// Handle active mutations
+	for(var/datum/mutation/mutation as anything in dna.mutations)
+		mutation.on_life(seconds_per_tick, times_fired)
+
+	// Heart attack stuff
+	handle_heart(seconds_per_tick, times_fired)
+	// Handles liver failure effects, if we lack a liver
+	handle_liver(seconds_per_tick, times_fired)
+	// For special species interactions
+	dna.species.spec_life(src, seconds_per_tick, times_fired)
+	return TRUE
```

Comment:
You cannot `return TRUE`, you have to check stat again. Any of the above procs could have killed the mob.